### DISCUSSION TITLE
Incorporation of Decay Amplitude

### DISCUSSION THREAD

**Discussion by mmikhasenko** [hep-statistics-standard/hep-statistics-standard#36] [Ideas]
As an outcome of today's discussion with @cburgard and @Cornelius-G
the way the hadronic models discussed in #34 and in [AmplitudeModelSerialization](https://github.com/mmikhasenko/AmplitudeModelSerialization).
would enter the HS3 is

In hadron physics we fit differential cross-section of decays, 1->N processes, or scattering, 2->N processes, described as 
```
dCrossSection(variables | parameters) = | M(variables|parameters) |^2
```
where `variables` parametrize N-body phase space using a recursive decomposition: N-body phase space is split into N-1 two-body vertices, each of which contribute with 3 variables, mass and spherical angles. There are observable (they vary event by event).
The matrix element `M` modelled by the pdf can also have `parameters` e.g. mass of a resonance in a particle pars.

```json
  "distributions": [
    {
      "name": "my_model_for_reaction_intensity",
      "type": "hadronic_cross_section_unpolarized_dist",
      "model_description": {
        "kinematics": {},
        "chains": [{}],
        "reference_topology": [[1, 2], 3],
        "text_replacement_blocks": {}
      },
      "variables": [
        {
          "node": [1, 2],
          "triplet": ["cos_theta_12", "phi_12", "m_12"]
        },
        {
          "node": [[1, 2], 3],
          "triplet": ["cos_theta_12_3", "phi_12_3", "m_12_3"]
        }
      ],
      "parameters": ["L_1520_mass"]
    }
  ],
```

Decomposition of the phase space variables is clarified with an example,
of a 3-body decay process described by the notation `[[1,2],3]`, for which nodes are indicated in the graph.
```
      0
      |
  [[1,2],3]
     /  \
    /    3
  [1,2]
  /  \
 /    2
1
```

The resonance lineshapes are introduced as `functions`. This should be a function of one of the mass variables, e.g. `m_12`.
It return a complex number, an amplutude, that is being used turned into a real PDF in `"hadronic_cross_section_unpolarized_dist"` model implementation.

```json
  "functions": [
    {
      "name": "L_1520_BW",
      "type": "complexBreitWigner",
      "mass": "L_1520_mass",
      "width": 1.52,
      "l": 2,
      "ma": 0.938,
      "mb": 0.548
    }
  ]
```

> [!WARNING]
> Does the variable of the function need to be specified?
> So far only parameters are given.

The `"domains"` section sets up intervals for the kinematic variables
as a product (`product_domain`) of 6 domains.

<details>
  <summary>See the `domains`</summary>

```json
  "domains": [
    {
      "name": "default",
      "type": "product_domain",
      "axes": [
        {
          "name": "cos_theta_12",
          "min": -1.0,
          "max": 1.0
        },
        {
          "name": "phi_12",
          "min": -3.14,
          "max": 3.14
        },
        {
          "name": "m_12",
          "min": 1.0,
          "max": 5.5
        },
        {
          "name": "cos_theta_12_3",
          "min": -1.0,
          "max": 1.0
        },
        {
          "name": "phi_12_3",
          "min": -3.14,
          "max": 3.14
        },
        {
          "name": "m_12_3",
          "min": 5.5,
          "max": 5.5
        },
        {
          "name": "L_1520_mass",
          "min": 1.45,
          "max": 1.6
        }
      ]
    }
  ],
```
</details>

Due to the complexity of the problem, I find it important to implement check sums from the start.
It's done via saving the expectation values of PDF for a given sample point in the file, `misc` section.

```json
  "misc": {
    "amplitude_model_checksums": [
      {
        "point": "validation_point",
        "distribution": "my_model_for_reaction_intensity",
        "value": 1.3
      }
    ]
  },
```

The section `"parameter_points"` gives default values to parameters, as well as the points sampled for validation.

<details>
  <summary>See the `parameter_points`</summary> 

```json
  "parameter_points": [
    {
      "name": "default_values_of_my_model",
      "parameters": [
        {
          "name": "L_1520_mass",
          "value": 4.6
        }
      ]
    },
    {
      "name": "validation_point",
      "parameters": [
        {
          "name": "L_1520_mass",
          "value": 4.6
        },
        {
          "name": "cos_theta_12",
          "value": 0.6
        },
        {
          "name": "phi_12",
          "value": 0.3
        },
        {
          "name": "m_12",
          "value": 3.6
        },
        {
          "name": "cos_theta_12_3",
          "value": 0.7
        },
        {
          "name": "phi_12_3",
          "value": -0.2
        },
        {
          "name": "m_12_3",
          "value": 5.6
        }
      ]
    }
  ]
```

</details>

**Comment by mmikhasenko** [comment]
Not clear why `"distributions"` need both `"variables"` and `"parameters"`, while functions do not.
@cburgard should variable / parameters be listed for every function?

**Reply by cburgard** [reply]
Hi Misha!
In principle, none of these are needed at all - you can create a distribution or a function without announcing the variables or parameters.
There is one consideration that might give an incentive to put variables or parameters somewhere easily accessible, and that is dependency tracking.
Any software actually trying to read one of these models will need to come up with an order in which it needs to create the objects, and that becomes easier if complicated objects (such as your decay amplitude distributions) announce the objects on which they depend in some easily accessible, top-level key.
But it's not required at all, so if this gives you a headache, you can drop this in either case, or make it optional.

**Comment by redeboer** [comment]
> ```json
>   "domains": [
>     {
>       "name": "default",
>       "type": "product_domain",
>       "axes": [
>         {
>           "name": "cos_theta_12",
>           "min": -1.0,
>           "max": 1.0
>         },
>         {
>           "name": "phi_12",
>           "min": -3.14,
>           "max": 3.14
>         },
>         {
>           "name": "m_12",
>           "min": 1.0,
>           "max": 5.5
>         },
>         {
>           "name": "cos_theta_12_3",
>           "min": -1.0,
>           "max": 1.0
>         },
>         {
>           "name": "phi_12_3",
>           "min": -3.14,
>           "max": 3.14
>         },
>         {
>           "name": "m_12_3",
>           "min": 5.5,
>           "max": 5.5
>         },
>         {
>           "name": "L_1520_mass",
>           "min": 1.45,
>           "max": 1.6
>         }
>       ]
>     }
>   ],
> ```

(Just posting while discussing with @mmikhasenko):
The mass domains cannot be factorised as a box, i.e. `m_12` currently has a fixed min-max, but is actually dependent on the other variables (`m_12_3`).
